src/Search/Annotation/GearmanTaskDriver.php

```diff
@@ -133,7 +137,7 @@ public function work()
         $this->logger->info('Worker started.');
         $this->addTasksToWorker($this->worker);
         $limit = $this->limit;
-        while (!$limit()) {
+        while (!$limit() && Signals::isValid()) {
```

Comment:
can you unify `Signals` with `MemoryLimit` in a Composite pattern so that here we can just do:
```
while ($limit()) {
    ...
}
```
where that object being injected and created in Kernel is:
```
new CompositeLimit([
    MemoryLimit::mb(...),
    SignalsLimit::sigterm(),
    ... // any other reason for stopping we add later
]);
```